src/lib/nvme_pt_ioctl.c

```diff
@@ -727,6 +733,128 @@ static int validate_sessn(struct opal_device *dev)
+{
+	struct opal_device *dev = device->priv;
+	int payload_len = dev->payload.len;
+	struct sed_tper_properties *tper = &device->discv.sed_tper_props;
+
+	/* TPer properties are returned as key-value pairs */
+	for (int i = 0; i < payload_len; i++) {
+		if(resp_token_match(dev->payload.tokens[i], OPAL_STARTNAME)) {
+			int jmp = get_payload_string(dev, i + 1);
+			int key_len = dev->payload.tokens[i + 1]->len - jmp;
+			char *key = malloc(key_len);
+			if (!key)
+				return -ENOMEM;
+
+			memset(key, 0, key_len);
+			memcpy(key, dev->payload.tokens[i + 1]->pos + jmp, key_len);
+
+			if (strncmp("MaxComPacketSize", key, key_len) == 0)
+				tper->MaxComPacketSize = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("MaxPacketSize", key, key_len) == 0)
+				tper->MaxPacketSize = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("MaxIndTokenSize", key, key_len) == 0)
+				tper->MaxIndTokenSize = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("MaxPackets", key, key_len) == 0)
+				tper->MaxPackets = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("MaxSubpackets", key, key_len) == 0)
+				tper->MaxSubpackets = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("MaxMethods", key, key_len) == 0)
+				tper->MaxMethods = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("MaxResponseComPacketSize", key, key_len) == 0)
+				tper->MaxResponseComPacketSize = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("MaxSessions", key, key_len) == 0)
+				tper->MaxSessions = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("MaxAuthentications", key, key_len) == 0)
+				tper->MaxAuthentications = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("MaxTransactionLimit", key, key_len) == 0)
+				tper->MaxTransactionLimit = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("DefSessionTimeout", key, key_len) == 0)
+				tper->DefSessionTimeout = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("MaxSessionTimeout", key, key_len) == 0)
+				tper->MaxSessionTimeout = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("MinSessionTimeout", key, key_len) == 0)
+				tper->MinSessionTimeout = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("DefTransTimeout", key, key_len) == 0)
+				tper->DefTransTimeout = dev->payload.tokens[i+2]->vals.uint;
+			else if (strncmp("MaxTransTimeout", key, key_len) == 0)
+				tper->MaxTransTimeout = dev->payload.tokens[i+2]->vals.uint;
+
+			free(key);
+		}
+	}
+
+	return 0;
+}
+
+static struct opal_req_item opal_tper_host_prep_cmd[] = {
+	{ .type = OPAL_U8, .len = 1, .val = { .byte = OPAL_STARTNAME } },
+	{ .type = OPAL_U8, .len = 1, .val = { .byte = OPAL_HOSTPROPERTIES } },
+	{ .type = OPAL_U8, .len = 1, .val = { .byte = OPAL_STARTLIST } },
+
+	{ .type = OPAL_U8, .len = 1, .val = { .byte = OPAL_STARTNAME } },
+	{ .type = OPAL_BYTES, .len = 16, .val = { .bytes = (uint8_t *)"MaxComPacketSize" } },
```

Comment:
After reading spec wording I think we should not communicate minimum HostProperties to TPer, because it will indicate to Tper that host can't do better than the minimums.
Our intention is to discover whatever TPer is capable of and update and update buffer size etc. to match what Tper advertises to us.

Here is the relevant spec wording:
The TPer uses these host properties when it is constructing responses to be transmitted to the host.
The host MAY omit properties as necessary, depending on the host’s communications capabilities. If
the host omits a property, the value of that property SHALL NOT change from its current value. If the
host specifies a value for a property that does not meet the minimum requirement as defined in Table
168, then the TPer SHALL use the minimum value defined in Table 168 in place of the value supplied
by the host

Reply:
Yeah. The maxcompacketsize can be removed off from here.

Reply:
we should remove HostProperties Parameter from this call not just single name-value pair

Reply:
I agree with that...getting rid of the Host Tper properties is a good idea.
From my interpretation of the spec lines you mentioned, if the HOST properties mentioned doesn't meet the minimum values as mentioned in the table 168, only then will the host properties value get substituted.